marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs

```diff
@@ -408,11 +413,30 @@ function versionIsCompatibleWithES(version = xdmp.version()) {
   return false;
 }
 
+function extractInstance(docNode) {
+  let instance = docNode.xpath('/*:envelope/(object-node("instance")|*:instance/(element() except *:info))');
```

Comment:
I think this warrants multiple unit tests directly on this function (in addition to the test you added already, which has a broader scope). The XPath expression is supporting multiple scenarios; we should have a unit test for each scenario that it supports to ensure that none of them break in the future, in case someone tries to change this.

Reply:
Added an additional test specifically for extractInstance. In the future I think it would be great to get set criteria for out unit tests. For example, are going for code coverage? Are going for direct calls to each function with a cyclomatic complexity of X or more? etc.